src/compile/Component.ts

```diff
@@ -1062,18 +1063,21 @@ export default class Component {
 						}
 
 						if (node.type === 'AssignmentExpression') {
-							const { name } = getObject(node.left)
-							assignees.add(name);
-							dependencies.delete(name);
+							const identifier = getObject(node.left)
+							assignee_nodes.add(identifier);
+							assignees.add(identifier.name);
 						} else if (node.type === 'UpdateExpression') {
-							const { name } = getObject(node.argument);
-							assignees.add(name);
-							dependencies.delete(name);
+							const identifier = getObject(node.argument);
+							assignee_nodes.add(identifier);
+							assignees.add(identifier.name);
 						} else if (isReference(node, parent)) {
-							const { name } = getObject(node);
-							const owner = scope.findOwner(name);
-							if ((!owner || owner === component.instance_scope) && (name[0] === '$' || component.var_lookup.has(name)) && !assignees.has(name)) {
-								dependencies.add(name);
+							const identifier = getObject(node);
+							if (!assignee_nodes.has(identifier)) {
+								const { name } = identifier;
+								const owner = scope.findOwner(name);
+								if ((!owner || owner === component.instance_scope) && (name[0] === '$' || component.var_lookup.has(name))) {
```

Comment:
Is now a good time to make sure the logic on this line makes sense? I get the `name[0] === '$' || component.var_lookup.has(name)` (this is a store subscription or it is not a global/unknown variable) but I'm not sure what the first half is checking for. What kind of scope stuff is that taking care of?

Reply:
The first half is checking that if you have something like this...

```js
let foo = 1;

$: {
  const answer = 42;
  console.log(answer);
  console.log(someGlobal);
  console.log(foo);
}
```

...only `foo` is treated as a dependency since `someGlobal` isn't declared in the script block (doesn't have an 'owner'), and `answer` is declared beneath the top level.

At least that's what I *think* it's doing

Reply:
Wouldn't `!owner` be true for `someGlobal`?

Reply:
It seems to be working correctly insofar as this...

``js
<script>
	let foo = 1;
	
	window.someGlobal = 'blah';

	$: {
		const answer = 42;
		console.log(answer);
		console.log(someGlobal);
		console.log(foo);
	}
</script>
```

...yields this:

```js
window.someGlobal = 'blah';

$$self.$$.update = ($$dirty = { console: 1, someGlobal: 1, foo: 1 }) => {
	if ($$dirty.foo) {
			const answer = 42;
			console.log(answer);
			console.log(someGlobal);
			console.log(foo);
		}
};
```

Though the `$$dirty = { console: 1, someGlobal: 1, foo: 1 }` could be improved

Reply:
Hmm even so, I guess this would be stopped probably by `component.var_lookup.has(name)`. I dunno.

Reply:
I'm prepared to file it under 'if it ain't broke' for now 😀 I reckon there's code that's in far more dire need of refactoring

Reply:
Yep sounds fine to me.